packages/scenario/test/select.test.ts

```diff
@@ -639,6 +642,131 @@ describe('SelectChoiceTest.java', () => {
+
+			const scenario = await Scenario.init('external-select-geojson.xml');
+
+			expect(scenario.choicesOf('/data/q').get(1)?.getChild('geometry')).toBe('0.5 104 0 0');
+			expect(scenario.choicesOf('/data/q').get(1)?.getChild('special-property')).toBe(
+				'special value'
+			);
+		});
+
+		it.fails(
+			'returns null when choices are from secondary instance and requested child does not exist',
+			async () => {
+				setUpSimpleReferenceManager(r('external-select-geojson.xml').getParent(), 'file');
+
+				const scenario = await Scenario.init('external-select-geojson.xml');
+
+				expect(scenario.choicesOf('/data/q').get(1)?.getChild('non-existent')).toBe(null);
+			}
+		);
+
+		it.fails(
+			'returns empty string when choices are from secondary instance and requested child has no value',
+			async () => {
+				const scenario = await Scenario.init(
+					'Select with empty value',
+					html(
+						head(
+							title('Select with empty value'),
+							model(
+								mainInstance(t("data id='select-empty'", t('select'))),
+								instance('choices', t('item', t('label', 'Item'), t('property', '')))
+							)
+						),
+						body(select1Dynamic('/data/select', "instance('choices')/root/item", 'name', 'label'))
+					)
+				);
+
+				expect(scenario.choicesOf('/data/select').get(0)?.getChild('property')).toBe('');
+			}
+		);
+
+		/**
+		 * **PORTING NOTES**
+		 *
+		 * This test is also blocked on lack of support for repeat-based itemsets.
+		 */
+		it.fails('updates when choices are from repeat', async () => {
+			const scenario = await Scenario.init(
+				'Select from repeat',
+				html(
+					head(
+						title('Select from repeat'),
+						model(
+							mainInstance(
+								t(
+									"data id='repeat-select'",
+									t('repeat', t('value'), t('label'), t('special-property')),
+									t('filter'),
+									t('select')
+								)
+							)
+						)
+					),
+					body(
+						repeat('/data/repeat', input('value'), input('label'), input('special-property')),
+						input('filter'),
+						select1Dynamic('/data/select', '../repeat')
+					)
+				)
+			);
+			scenario.answer('/data/repeat[0]/value', 'a');
+			scenario.answer('/data/repeat[0]/label', 'A');
+			scenario.answer('/data/repeat[0]/special-property', 'AA');
+
+			expect(scenario.choicesOf('/data/select').get(0)?.getValue()).toBe('a');
+			expect(scenario.choicesOf('/data/select').get(0)?.getChild('special-property')).toBe('AA');
+
+			scenario.answer('/data/repeat[0]/special-property', 'changed');
+
+			expect(scenario.choicesOf('/data/select').get(0)?.getChild('special-property')).toBe(
+				'changed'
+			);
+		});
+
+		/**
+		 * **PORTING NOTES**
+		 *
+		 * In theory, this could be made to pass! It makes more sense to fail it
+		 * with the same error as the others above, as it is also subject to the
+		 * same API design considerations. It also may be moot depending on our
+		 * posture towards inline select items generally.
+		 */
```

Comment:
By "In theory, this could be made to pass!" what I mean is that it would be possible to implement the `SelectChoice.getChild` method such that it always returns `null` for now instead of throwing/triggering immediate test failure. I think the future of this test's pertinence is probably more contingent on whatever implications this has for client-facing APIs (and apparently potentially overlapping with `appearance` as noted in the  thread immediately above). Whatever we do here will probably work regardless of whether selects are inline, they're already roughly at parity with itemsets and I don't see that changing unless we explicitly **deprioritize that** (which it sounds like we won't).